app/src/main/java/org/permanent/permanent/repositories/ProfileRepositoryImpl.kt

```diff
@@ -0,0 +1,38 @@
+package org.permanent.permanent.repositories
+
+import android.content.Context
+import org.permanent.permanent.network.IDataListener
+import org.permanent.permanent.network.NetworkClient
+import org.permanent.permanent.network.models.ResponseVO
+import org.permanent.permanent.ui.PREFS_NAME
+import org.permanent.permanent.ui.PreferencesHelper
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class ProfileRepositoryImpl(val context: Context) : IProfileRepository {
+    private val prefsHelper = PreferencesHelper(
+        context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
+    )
+
+    override fun getProfileItemsByArchive(listener: IDataListener) {
+        NetworkClient.instance()
+            .getProfileItemsByArchive(prefsHelper.getCsrf(), prefsHelper.getCurrentArchiveNr())
```

Comment:
the method expects byArchive but no archive is specified in the signature but only in the implementation